Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,411 @@
+            foreach (var teamID in m.TeamIDs)
+            {
+                await CreateMatchTeamMappingAsync(teamID, match.ID);
+            }
+            await _context.SaveChangesAsync();
+            return match;
+        }
+
+        public async Task CreateMatchTeamMappingAsync(int teamID, int matchID)
+        {
+            var matchTeam = new MatchTeam
+            {
+                TeamID = teamID,
+                MatchID = matchID,
+                StatusID = 2, //default confirmed
+                Score = 0,
+                Sportsmanship = 5 //default max
+            };
+            await _context.MatchTeam.AddAsync(matchTeam);
+        }
+        public async Task<MatchTeamViewModel> UpdateTeamStatsAsync(int matchID, MatchStatsPatchViewModel vm)
+        {
+            var teamstats = _context.MatchTeam.FirstOrDefault(mt => mt.MatchID == matchID && mt.TeamID == vm.TeamID);
+            teamstats.Score = vm.Score ?? teamstats.Score;
+            teamstats.Sportsmanship = vm.Sportsmanship ?? teamstats.Sportsmanship;
+            teamstats.StatusID = vm.StatusID ?? teamstats.StatusID;
+            await _context.SaveChangesAsync();
+            return teamstats;
+
+        }
+        public async Task<MatchViewModel> UpdateMatchAsync(int matchID, MatchPatchViewModel vm)
+        {
+            var match = _context.Match.FirstOrDefault(m => m.ID == matchID);
+            match.Time = vm.Time ?? match.Time;
+            match.StatusID = vm.StatusID ?? match.StatusID;
+            match.SurfaceID = vm.SurfaceID ?? match.SurfaceID;
+
+            if (vm.TeamIDs is not null)
+            {
+                List<int> updatedTeams = vm.TeamIDs.ToList();
+                var removedTeams = _context.MatchTeam.Where(mt => mt.MatchID == matchID && !updatedTeams.Any(t_id => mt.TeamID == t_id));
+                _context.MatchTeam.RemoveRange(removedTeams);
+
+                var existingTeams = _context.MatchTeam.Where(mt => mt.MatchID == matchID && updatedTeams.Any(t_id => mt.TeamID == t_id));
+                var teamsToAdd = updatedTeams.Where(id => !existingTeams.Any(t => t.TeamID == id));
+                foreach (int id in teamsToAdd)
+                {
+                    await CreateMatchTeamMappingAsync(id, matchID);
+                }
+            }
+
+            await _context.SaveChangesAsync();
+            return match;
+        }
+
+        public async Task DeleteMatchCascadeAsync(int matchID)
+        {
+            //delete matchteam
+            var matchteam = _context.MatchTeam.Where(mt => mt.MatchID == matchID);
+            _context.MatchTeam.RemoveRange(matchteam);
+            //delete matchparticipant
+            var matchparticipant = _context.MatchParticipant.Where(mp => mp.MatchID == matchID);
+            _context.MatchParticipant.RemoveRange(matchparticipant);
+            //deletematch
+            var match = _context.Match.FirstOrDefault(m => m.ID == matchID);
+            _context.Match.Remove(match);
```

Comment:
Since this is doing a true delete, it looks like a great use case for `on delete cascade`, so that when you delete the Match, all the related MatchTeam and MatchParticipant rows get deleted automatically.

Reply:
We actually implemented that in SQL just after I finished the routes here. After which I updated to "fake" delete with a deleted status.